lib/suppression_list.ex

```diff
@@ -1,18 +1,69 @@
 defmodule SparkPost.SuppressionList do
   @moduledoc """
   The SparkPost Suppression List API for working with suppression lists.
-  Use `SparkPost.SuppressionList.search/1` to search through your account's suppression list.
+  Use `SparkPost.SuppressionList.delete/1` to delete a single entry from a list,
+  `SparkPost.SuppressionList.update_one/3` to insert or update a single list entry,
+  or `SparkPost.SuppressionList.search/1` to search through your account's suppression list.
 
   Check out the documentation for each function
   or use the [SparkPost API reference](https://developers.sparkpost.com/api/suppression_list.html) for details.
 
+  Returned by `SparkPost.SuppressionList.delete/1`:
+    - nothing (empty body)
+
+  Returned by `SparkPost.SuppressionList.update_one/3`:
+    - message (A success message string)
+
   Returned by `SparkPost.SuppressionList.search/1`.
     - %SparkPost.SuppressionList.SearchResult{}
   """
 
   alias SparkPost.Endpoint
 
   @doc """
+  Insert or update a single entry in the suppression list.
+  Returns a single string with the success message if the entry
+  was updated or inserted. Returns a %SparkPost.Endpoint.Error{} with a 400
+  if the type is invalid.
+
+  Parameters:
+    - recipient: the email to insert or update in the suppression list
+    - type: one of "transactional" or "non_transactional"
+    - description (optional): optional description of this entry in the suppression list
+  """
+  def update_one(recipient, type, description \\ nil) do
+    body = if description == nil do
+      %{type: type}
+    else
+      %{type: type, description: description}
+    end
+    response = Endpoint.request(:put, "suppression-list/#{recipient}", body)
```

Comment:
I'd like to adopt this. Unfortunately I'm not sure I have enough time to go through and update the library everywhere to make sure everything still works. Might make sense to have this in a different PR?